OS::is_process_running function.

mdavisprog opened the pull request:
This PR adds the OS::is_process_running function that is exposed to script. The function has been implemented for Windows and Unix platforms and provides stubs for other platforms that are unsupported.

This is relevant to the godot-proposal [#1679](https://github.com/godotengine/godot-proposals/issues/1679).

A test project is provided that creates a process to run the 'ping' command and asserts that the is_process_running function properly returns the correct value.

[os-is-process-running-project.zip](https://github.com/godotengine/godot/files/6987421/os-is-process-running-project.zip)

*Bugsquad edit: This closes https://github.com/godotengine/godot-proposals/issues/1679.*

mdavisprog committed:
Add OS::is_process_running function.

Adds the is_process_running function to the native OS class and exposes it to script.

This is implemented on Windows and Unix platforms. A stub is provided for other platforms that do not support this function.

Documentation is updated to reflect new API function.

Ev1lbl0w commented:
Came looking for this, this is very much needed for a project I'm working on, any chance of this being merged?

akien-mga commented on platform/javascript/os_javascript.cpp:
```diff
@@ -132,6 +132,10 @@ int OS_JavaScript::get_process_id() const {
 }
 
+bool OS_JavaScript::is_process_running(const ProcessID &p_pid) const {
+	ERR_FAIL_V_MSG(ERR_UNAVAILABLE, "OS::is_process_running() is not available on the HTML5 platform.");
```
This should return `false`, not `ERR_UNAVAILABLE`.

akien-mga commented on platform/windows/os_windows.cpp:
```diff
@@ -507,6 +507,22 @@ int OS_Windows::get_process_id() const {
 }
 
+bool OS_Windows::is_process_running(const ProcessID &p_pid) const {
+	if (!process_map->has(p_pid))
+		return false;
```
Please use braces even for single line `if` statements (same below).

mdavisprog commented:
Hello,

I have implemented the feedback and the branch is now rebased with the latest from master. If there is anything else that needs attention, let me know.

akien-mga commented:
Thanks!

MikeSchulze commented:
thanks

akien-mga commented:
Cherry-picked for 3.5.
